packages/nuxt/src/templates/plugin.ts

```diff
@@ -43,10 +48,14 @@ const PiniaNuxtPlugin: Plugin = (context, inject) => {
       })
     } else {
       // there is no beforeNuxtRender in Nuxt 3
-      context.nuxtState.pinia = pinia.state.value
+      // @ts-expect-error: vue 3 only
+      context.ssrContext.payload.pinia = pinia.state.value
+    }
+  } else {
+    const source = isVue2 ? context.nuxtState : context.payload
```

Comment:
Was this change necessary? 🤔 
It created other problems: https://github.com/vuejs/pinia/issues/1447

Reply:
Because nuxtState is removed in Nuxt 3 https://github.com/nuxt/framework/pull/5630

It's weird that I didn't get error when set `ssr: false`.

Reply:
ok, thanks